cluster/cluster_config.go

```diff
@@ -21,89 +21,83 @@ import (
-		SmartRouting:         true,
-		ConnectionTimeout:    5 * time.Second,
-		HeartbeatInterval:    5 * time.Second,
-		HeartbeatTimeout:     60 * time.Second,
-		InvocationTimeout:    120 * time.Second,
-		SecurityConfig:       NewSecurityConfig(),
-		SSLConfig:            NewSSLConfig(),
-		HazelcastCloudConfig: NewHazelcastCloudConfig(),
-		DiscoveryConfig:      NewDiscoveryConfig(),
-	}
+type Config struct {
+	Security          SecurityConfig       `json:",omitempty"`
+	SSL               SSLConfig            `json:",omitempty"`
+	Name              string               `json:",omitempty"`
+	HazelcastCloud    HazelcastCloudConfig `json:",omitempty"`
+	Address           []string             `json:",omitempty"`
+	InvocationTimeout types.Duration       `json:",omitempty"`
+	HeartbeatInterval types.Duration       `json:",omitempty"`
+	HeartbeatTimeout  types.Duration       `json:",omitempty"`
+	ConnectionTimeout types.Duration       `json:",omitempty"`
+	Discovery         DiscoveryConfig      `json:",omitempty"`
+	RedoOperation     bool                 `json:",omitempty"`
+	Unisocket         bool                 `json:",omitempty"`
 }
 
 func (c *Config) Clone() Config {
-	addrs := make([]string, len(c.Addrs))
-	copy(addrs, c.Addrs)
+	addrs := make([]string, len(c.Address))
+	copy(addrs, c.Address)
 	return Config{
-		Name:                 c.Name,
-		Addrs:                addrs,
-		SmartRouting:         c.SmartRouting,
-		ConnectionTimeout:    c.ConnectionTimeout,
-		HeartbeatInterval:    c.HeartbeatInterval,
-		HeartbeatTimeout:     c.HeartbeatTimeout,
-		InvocationTimeout:    c.InvocationTimeout,
-		RedoOperation:        c.RedoOperation,
-		SecurityConfig:       c.SecurityConfig.Clone(),
-		SSLConfig:            c.SSLConfig.Clone(),
-		HazelcastCloudConfig: c.HazelcastCloudConfig.Clone(),
-		DiscoveryConfig:      c.DiscoveryConfig.Clone(),
+		Name:              c.Name,
+		Address:           addrs,
+		Unisocket:         c.Unisocket,
+		ConnectionTimeout: c.ConnectionTimeout,
+		HeartbeatInterval: c.HeartbeatInterval,
+		HeartbeatTimeout:  c.HeartbeatTimeout,
+		InvocationTimeout: c.InvocationTimeout,
+		RedoOperation:     c.RedoOperation,
+		Security:          c.Security.Clone(),
+		SSL:               c.SSL.Clone(),
+		HazelcastCloud:    c.HazelcastCloud.Clone(),
+		Discovery:         c.Discovery.Clone(),
 	}
 }
 
 func (c *Config) Validate() error {
 	if c.Name == "" {
-		return ErrConfigInvalidClusterName
+		c.Name = defaultName
 	}
-	for _, addr := range c.Addrs {
-		if err := checkAddress(addr); err != nil {
-			return fmt.Errorf("invalid address %s: %w", addr, err)
+	if len(c.Address) == 0 {
+		c.Address = []string{defaultAddress}
+	} else {
+		for _, addr := range c.Address {
+			if err := checkAddress(addr); err != nil {
+				return fmt.Errorf("invalid address %s: %w", addr, err)
+			}
 		}
 	}
-	if c.ConnectionTimeout < 0 {
-		return ErrConfigInvalidConnectionTimeout
+	if c.ConnectionTimeout <= 0 {
```

Comment:
Just curious, is the zero value of connection timeout a special value in Go?

Reply:
Zero value of integers is `0`.